AppServer/google/appengine/tools/appengine_rpc.py

```diff
@@ -521,8 +522,10 @@ def Send(self, request_path, payload="",
     if 'AUTH_DOMAIN' in os.environ:
       auth_domain = os.environ['AUTH_DOMAIN'].lower()
 
+    opener_has_timeout = 'timeout' in inspect.getargspec(self.opener.open).args
     old_timeout = socket.getdefaulttimeout()
```

Comment:
I agree. Unfortunately, PyCharm complains that `old_timeout` might be referenced before assignment in the finally block. I don't know the best way to organize it without adding a warning.